### PR TITLE
[OBSDEF-5917] Use dedicated service account for atlas-operator

### PR DESCRIPTION
**Pull request by markadev**
## Purpose
[OBSDEF-5917](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-5917)

This PR updates the atlas-operator chart to create a dedicated ServiceAccount for the atlas-operator pod instead of reusing the default service account.

## PR checklist
- [x] make test passed
- [x] make build passed
- [ ] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed

## Testing

https://asd-ecs-jenkins.isus.emc.com/jenkins/job/charts-custom-ci/72/

### CODE REVIEW

File: atlas-operator/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{{- if .Values.serviceAccount.create }}
```
Comment (vgupta-mickey) [inline atlas-operator/templates/serviceaccount.yaml:1]: I don't think that we should have an option to create service account. We should always create.

File: atlas-operator/templates/role_binding.yaml [hunk 2]
```diff
@@ -6,12 +6,12 @@ metadata:
...
-{{ include "atlas-operator.labels" . | indent 4 }}
+    {{- include "atlas-operator.labels" . | nindent 4 }}
 roleRef:
   apiGroup: rbac.authorization.k8s.io
   kind: {{ .Values.global.watchAllNamespaces | ternary "ClusterRole" "Role" }}
   name: atlas-operator
 subjects:
 - kind: ServiceAccount
-  name: default
+  name: {{ .Values.serviceAccount.name }}
```
Comment (vgupta-mickey) [inline atlas-operator/templates/role_binding.yaml:16]: I wouldn't allow customer change the service account name, so I wouldn't allow the name of the account to be configurable.

File: atlas-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -17,6 +17,7 @@ spec:
         app.kubernetes.io/name: {{ include "atlas-operator.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
     spec:
+      serviceAccountName: {{ .Values.serviceAccount.name }}
```
Comment (vgupta-mickey) [inline atlas-operator/templates/deployment.yaml:20]: I would hard code the service account name here such as "atlas-operator".

File: atlas-operator/templates/role_binding.yaml (hunk 2 above)
Comment (vgupta-mickey) [reply atlas-operator/templates/role_binding.yaml:16]: Also if create = false, then this will be invalid.

File: atlas-operator/values.yaml [hunk 4]
```diff
@@ -25,6 +25,10 @@ image:
 
 nameOverride: ""
 
+serviceAccount:
```
Comment (vgupta-mickey) [inline atlas-operator/values.yaml:28]: I would remove this from values.yaml as I don't think that we should give control to change the account name.

**Comment by markadev** [comment]
OK @vgupta-mickey, I addressed your comments

**Review by vgupta-mickey — APPROVED** [review]

**Review by eddiepavkovic — APPROVED** [review]
ok

**Pull request merged**